code/game/machinery/bots/medbot.dm

```diff
@@ -472,47 +472,58 @@
 			spawn(INJECTION_TIME)
 				inject_patient()
 
-//	src.speak(reagent_id)
-	reagent_id = null
-	return
-
 /obj/machinery/bot/medbot/proc/inject_patient()
-	if ((get_dist(src, src.patient) <= 1) && (src.on))
-		if((reagent_id == "internal_beaker") && (src.reagent_glass) && (src.reagent_glass.reagents.total_volume))
-			src.reagent_glass.reagents.trans_to(src.patient,src.injection_amount) //Inject from beaker instead.
-			src.reagent_glass.reagents.reaction(src.patient, 2)
+	var/succesful_inject = 0
+	if ((get_dist(src, patient) <= 1) && (on))
+		if((reagent_id == "internal_beaker") && (reagent_glass) && (reagent_glass.reagents.total_volume))
+			reagent_glass.reagents.trans_to(patient,injection_amount) //Inject from beaker instead.
+			reagent_glass.reagents.reaction(patient, 2)
+			succesful_inject = 1
 		else
-			src.patient.reagents.add_reagent(reagent_id,src.injection_amount)
-		visible_message("<span class='danger'>[src] injects [src.patient] with the syringe!</span>")
-
-	src.icon_state = "[src.icon_initial][src.on]"
-	src.currently_healing = 0
+			if(!patient.reagents.has_reagent(reagent_id) && !emagged) //Somebody got there first
```

Comment:
Primary change three - Now it double checks if the patient has the reagent it's going to inject into said patient. First when evaluating, and secondly just before injection.

If the patient does have the chemical, it does not return a succesful injection, and therefore does not print that the patient was injected with the syringe